wallet/chainntfns.go

```diff
@@ -174,6 +181,44 @@ func (w *Wallet) ChainSwitch(ctx context.Context, forest *SidechainForest, chain
 
 			// Add the block hash to the notification.
 			chainTipChanges.AttachedBlocks = append(chainTipChanges.AttachedBlocks, n.Hash)
+
+			if birthdayState != nil && (birthdayState.SetFromHeight || birthdayState.SetFromTime) {
+				if birthdayState.SetFromTime && n.Header.Timestamp.After(birthdayTime) {
+					bh := n.Header.PrevBlock
+					birthdayState.Hash = bh
+					birthdayState.Height = n.Header.Height - 1
+					birthdayState.SetFromTime = false
+					if err := w.txStore.SetBirthState(dbtx, birthdayState); err != nil {
+						return err
+					}
+					if err := w.txStore.UpdateProcessedTxsBlockMarker(dbtx, &bh); err != nil {
+						return err
+					}
+					log.Infof("Set wallet birthday to block %d (%v).",
+						n.Header.Height-1, bh)
+				} else if birthdayState.SetFromHeight && n.Header.Height == birthdayState.Height+1 {
+					bh := n.Header.PrevBlock
+					birthdayState.Hash = bh
+					birthdayState.SetFromHeight = false
+					if err := w.txStore.SetBirthState(dbtx, birthdayState); err != nil {
+						return err
+					}
+					if err := w.txStore.UpdateProcessedTxsBlockMarker(dbtx, &bh); err != nil {
+						return err
+					}
+					log.Infof("Set wallet birthday to block %d (%v).",
+						n.Header.Height-1, bh)
+				} else if i == len(chain)-1 {
+					// In the case that the birthday will not be reached,
+					// store a hash for initial sync to start from. Do not
+					// store the tip hash as that will cause w.rescanPoint
+					// to return nil.
+					bh := n.Header.PrevBlock
+					if err := w.txStore.UpdateProcessedTxsBlockMarker(dbtx, &bh); err != nil {
+						return err
+					}
```

Comment:
I'm not sure you need this case. At this point in the code, you know the birthday will bet set at some point in the future (in which case one of the prior branches will be reached), so what's the need to do this for every batch of blocks until you reach the birthday height?

Reply:
If the birthday is set in the future then, although we don't need to, we will scan from genesis.

I will remove the check anyway though since this is user error I suppose. If the birthday is set to a time or block in the future, it is not fatal.